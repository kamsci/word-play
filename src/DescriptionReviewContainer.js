import React, { Component } from 'react';
import DescriptionReviewCard from './DescriptionReviewCard';
import dictionary from '../dictionary.json';

class DescriptionReviewContainer extends Component {
  /* Container class to handle source of data*/

  constructor(props) {
    super(props);

    this.state = {
      bool: props.bool,
      title: props.title,
      description: this.inputStringToSpanArray(props.description)
    };
  }

  inputStringToSpanArray = (input) => {
    // Parse paragraph into array
    let inputArr = input.split(" ");
    let descriptionArr = [];
    let arrayCount = 0;
    
    for(var i =0; i <inputArr.length; i++) {
      let tempPhrase2 = inputArr[i] + " " + inputArr[i+1];
      let tempPhrase3 = inputArr[i] + " " + inputArr[i+1] + " " + inputArr[i+2];
      let id = "Word" + arrayCount;

        // if single key word
      if(this.isKeyWord(inputArr[i]) === true) {
        descriptionArr.push(<span id={id} className="match" key={arrayCount}>{inputArr[i]}</span>)
      } // if 2 word phrase
      else if (this.isKeyWord(tempPhrase2) === true) {
        descriptionArr.push(<span id={id} className="match" key={arrayCount}>{tempPhrase2}</span>)
        i++;
      } // if 3 word phrase
      else if (this.isKeyWord(tempPhrase3) === true) {
        descriptionArr.push(<span id={id} className="match" key={arrayCount}>{tempPhrase3}</span>)
        i += 2;
      } 
      else {
        descriptionArr.push(inputArr[i]);
      }
      arrayCount++;
    }
    return descriptionArr;
  }
  
  // Identify if key word/phrase is in dictionary
  isKeyWord = (word) => {
    word = word.toLowerCase();
    word = word.replace(/(,|!|\?|:|\.)/g, "");

    if(dictionary.hasOwnProperty(word)) { return true; }
    return false;
  }
  // create job description array without span elements
  inputStringToArrayWithoutSpans = (input) => {
    let inputArr = input.split(" ");
    let descriptionArr = [];
    
    for(var i =0; i <inputArr.length; i++) {
      let tempPhrase2 = inputArr[i] + " " + inputArr[i+1];
      let tempPhrase3 = inputArr[i] + " " + inputArr[i+1] + " " + inputArr[i+2];

        // if single key word
      if(this.isKeyWord(inputArr[i]) === true) {
        descriptionArr.push(inputArr[i]);
      } // if 2 word phrase
      else if (this.isKeyWord(tempPhrase2) === true) {
        descriptionArr.push(tempPhrase2);
        i++;
      } // if 3 word phrase
      else if (this.isKeyWord(tempPhrase3) === true) { 
        descriptionArr.push(tempPhrase3);
        i += 2;
      } 
      else {
        descriptionArr.push(inputArr[i]);
      }
    }
    return descriptionArr;
  }

  // replace new word with old word in job description string
  passToJobReviewCard = (wordState) => {
    let tempDescription = this.inputStringToArrayWithoutSpans(this.props.description);
    let word = wordState.props.children;
    tempDescription[wordState.key] = word;
    tempDescription = tempDescription.join(" ");

    this.props.passToJobReviewCard(tempDescription);
  }

  componentWillReceiveProps = (nextProps) => {
    this.setState({ 
      bool: nextProps.bool,
      title: nextProps.title,
      description: this.inputStringToSpanArray(nextProps.description)
    });
  }
  
  render() {
    return <DescriptionReviewCard  title={this.state.title} 
                                   description={this.state.description}
                                   passToDescriptionReviewContainer={this.passToJobReviewCard} />;
  }
}

export default DescriptionReviewContainer;