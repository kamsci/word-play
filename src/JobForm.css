.form {
  /*width: 70%;*/
  margin-top: 50px;
  margin-bottom: 50px;
  padding: 5px 50px;
}
#description {
  min-height: 300px;
}
.btn {
  background-color: #97c728;
  color: #fff;
}