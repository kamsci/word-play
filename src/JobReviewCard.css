.jobDescription {
  width: 75%;
  margin: auto auto;
  min-height: 550px;
  color: #6c696a;
}
.intro {
  text-align: center;
  margin-bottom: 20px;
  color: #565354;
}
.intro p {
  font-size: 18px;
}
.tip {
  margin-bottom: 30px;
}

/* On JobForm & DescriptionReviewCard */
.descriptionBox {
  margin-top: 0px;
  margin-bottom: 30px;
  font-size: 16px;
  line-height: 28px;
  min-height: 300px;
}