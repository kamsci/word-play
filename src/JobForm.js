import React, { Component } from 'react';
import './JobForm.css';

///////////////////////////////////////////

class JobForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: "",
      description: ""
    }

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
  }

  handleTitleChange = (event) => {
    // console.log("submit title", event)
    this.setState({
      title: event.target.value
    })
  }
  handleDescriptionChange = (event) => {
    // console.log("submit description", event)
    this.setState({
      description: event.target.value
    })
  }
  
  submitForm = (event) => {
    event.preventDefault();
    console.log("submit", this.state.title, this.state.description)
  }

  render() {
    return(
      <div className="continer form">
        <form onSubmit={this.submitForm}>
          <div className="form-group">
            <input id="title" 
                   type="text"
                   value={this.state.title}
                   onChange={this.handleTitleChange} 
                   className="form-control" 
                   placeholder="Job Title" />
          </div>
          <div className="form-group">
            <textarea id="description" 
                      type="text"
                      value={this.state.description}
                      onChange={this.handleDescriptionChange} 
                      className="form-control" 
                      placeholder="Job Descrption paragraph - no list items"/>
          </div>
          <input type="submit" className="btn" value="Submit" />
        </form>
      </div>
    )
  }
  
}

export default JobForm;