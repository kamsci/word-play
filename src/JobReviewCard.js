import React, { Component } from 'react';
//import './JobReviewCard.css';
import SentenceReviewCard from './SentenceReviewCard';
import DescriptionReviewCard from './DescriptionReviewCard';
import JobForm from './JobForm';

//////////////////////////////////////////

class JobReviewCard extends Component {

  render() {
    return (
      <div>
        <p className="intro">Click on one of the alternative words to replace the highlighted word in your sentence:</p>
        <SentenceReviewCard />
        <div className="container jobDescription">
          <DescriptionReviewCard />
          <JobForm />
        </div>
      </div>

    );
  }
}

export default JobReviewCard;