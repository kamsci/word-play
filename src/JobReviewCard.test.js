import React from 'react';
import ReactDOM from 'react-dom';
import JobReviewCard from './JobReviewCard';
import DescriptionReviewContainer from './DescriptionReviewContainer';
import JobForm from './JobForm';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<JobReviewCard />, div);
});

test('JobReviewCard changes when editMode toggles', () => {
  const component = renderer.create(
    <JobReviewCard />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger editMode
  tree.props.editMode = false;
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.editMode = true;
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});