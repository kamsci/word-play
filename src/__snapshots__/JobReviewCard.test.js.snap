exports[`test JobReviewCard changes when editMode toggles 1`] = `
<div
  className="container jobDescription">
  <div>
    <p
      className="intro">
      Submit a job description and we\'ll provide some alternative word suggestions.
    </p>
    <form
      onSubmit={[Function]}>
      <div
        className="form-group">
        <input
          className="form-control"
          id="title"
          onChange={[Function]}
          placeholder="Job Title"
          required={true}
          type="text"
          value="" />
      </div>
      <div
        className="form-group">
        <textarea
          className="form-control descriptionBox"
          id="description"
          onChange={[Function]}
          placeholder="Job Descrption paragraph - no list items"
          required={true}
          type="text"
          value="" />
      </div>
      <input
        className="btn"
        type="submit"
        value="Submit" />
    </form>
  </div>
</div>
`;

exports[`test JobReviewCard changes when editMode toggles 2`] = `
<div
  className="container jobDescription">
  <div>
    <p
      className="intro">
      Submit a job description and we\'ll provide some alternative word suggestions.
    </p>
    <form
      onSubmit={[Function]}>
      <div
        className="form-group">
        <input
          className="form-control"
          id="title"
          onChange={[Function]}
          placeholder="Job Title"
          required={true}
          type="text"
          value="" />
      </div>
      <div
        className="form-group">
        <textarea
          className="form-control descriptionBox"
          id="description"
          onChange={[Function]}
          placeholder="Job Descrption paragraph - no list items"
          required={true}
          type="text"
          value="" />
      </div>
      <input
        className="btn"
        type="submit"
        value="Submit" />
    </form>
  </div>
</div>
`;

exports[`test JobReviewCard changes when editMode toggles 3`] = `
<div
  className="container jobDescription">
  <div>
    <p
      className="intro">
      Submit a job description and we\'ll provide some alternative word suggestions.
    </p>
    <form
      onSubmit={[Function]}>
      <div
        className="form-group">
        <input
          className="form-control"
          id="title"
          onChange={[Function]}
          placeholder="Job Title"
          required={true}
          type="text"
          value="" />
      </div>
      <div
        className="form-group">
        <textarea
          className="form-control descriptionBox"
          id="description"
          onChange={[Function]}
          placeholder="Job Descrption paragraph - no list items"
          required={true}
          type="text"
          value="" />
      </div>
      <input
        className="btn"
        type="submit"
        value="Submit" />
    </form>
  </div>
</div>
`;
